q2_demux/_summarize/assets/src/plot.js

```diff
@@ -0,0 +1,229 @@
+
+  table
+    .append('thead')
+    .append('tr')
+      .selectAll('th')
+    .data([['Hover over a boxplot to learn more...', 9], ['', 3]])
+      .enter()
+    .append('th')
+    .text(d => d[0])
+    .attr('class', d => `col-xs-${d[1]}`)
+    .style('font-size', '10px');
+
+  const initialData = [
+    ['Position Number', '...'],
+    ['Minimum', '...'],
+    ['1st Quartile', '...'],
+    ['Median', '...'],
+    ['3rd Quartile', '...'],
+    ['Maximum', '...'],
+  ];
+
+  const rows = table
+    .append('tbody')
+      .selectAll('tr')
+      .data(initialData)
+    .enter()
+      .append('tr');
+
+  rows
+    .selectAll('td')
+    .data(d => d)
+      .enter()
+    .append('td')
+    .text(d => d);
+
+  const maxX = d3.max(data, d => d[0]);
+  const x0 = [0, maxX];
+  const y0 = [0, 100];
+  const x = d3.scaleLinear()
+    .domain(x0)
+    .range([props.margin.left, props.width]);
+  const y = d3.scaleLinear()
+    .domain(y0)
+    .range([props.height - props.margin.bottom, props.margin.top]);
+
+  const xAxis = d3.axisBottom(x).ticks(12);
+  const yAxis = d3.axisLeft(y).ticks((12 * props.height) / props.width);
+
+  const iqr = (d) => {
+    const q1 = d.quartiles[0];
+    const q3 = d.quartiles[2];
+    const iq = (q3 - q1) * 1.5;
+    let i = -1;
+    let j = d.length;
+    while (d[++i] < q1 - iq);  // eslint-disable-line no-plusplus
+    while (d[--j] > q3 + iq);  // eslint-disable-line no-plusplus
+    return [i, j];
+  };
+
+  const chart = d3.box()
+    .whiskers(iqr)
+    .height(props.height - props.margin.bottom - props.margin.top)
+    .domain([0, 100]);
+
+  const brush = d3.brush();
+  let idleTimeout;
+  const idleDelay = 350;
+
+  const idled = () => {
+    idleTimeout = null;
+  };
+
+  const zoom = () => {
+    const t = svg.transition().duration(750);
+    svg.select('.axis--x').transition(t).call(xAxis);
+    svg.select('.axis--y').transition(t).call(yAxis);
+    svg.selectAll('.boxplot').transition(t)
+      .call(chart.domain(y.domain())
+                 .width((x.range()[1] - x.range()[0]) / (x.domain()[1] - x.domain()[0]) / 2))
+      .attr('transform', d => `translate(${x(d[0])}, ${props.margin.top})`);
+  };
+
+  const brushEnded = () => {
+    const s = d3.event.selection;
+    if (!s) {
+      if (!idleTimeout) {
+        idleTimeout = setTimeout(idled, idleDelay);
+        return idleTimeout;
+      }
+      x.domain(x0);
+      y.domain(y0);
+    } else {
+      x.domain([s[0][0], s[1][0]].map(x.invert, x));
+      y.domain([s[1][1], s[0][1]].map(y.invert, y));
+      svg.select('.brush').call(brush.move, null);
+    }
+    return zoom();
+  };
+
+  brush.on('end', brushEnded);
+
+  svg.append('g')
+      .attr('class', 'brush')
+      .attr('z-index', -1)
+      .call(brush);
+
+  svg.selectAll('.box')
+    .attr('font', '10px sans-serif')
+    .data(data)
+  .enter()
+    .append('g')
+    .attr('class', 'boxplot')
+    .attr('transform', d => `translate(${x(d[0])}, ${props.margin.top})`)
```

Comment:
Oh, I see. `g`'s are only positionable using the `transform/translate` capabilities. `x` and `y` don't affect them. Using the `g`'s for each boxplot allows relative placement for the svg elements themselves, so they can be placed without even knowing their actual location, so the `x1`, `x2`, etc... in the plots can just be 0 instead of needing the axis to calculate anything.